engine/core/src/main/java/org/datacleaner/job/CompoundComponentRequirement.java

```diff
@@ -95,12 +130,21 @@ public boolean hasMultipleRequirementsFrom(HasFilterOutcomes producingComponent)
 
     @Override
     public boolean isSatisfied(InputRow row, FilterOutcomes outcomes) {
-        for (FilterOutcome outcome : outcomes.getOutcomes()) {
-            if (_outcomes.contains(outcome)) {
-                return true;
+        if(_compoundingType == CompoundingType.ANY || row == null) {
```

Comment:
Reviving your comment here (was removed due to a commit I guess)...

The ```row == null``` is in deed a little worrying. I see that when I remove it, I get exceptions in the ```PreviewTransformedDataActionListenerTest``` tests because DC's engine cannot calculate a proper processing order. Wondering what that means - give me 10 minutes to ponder on it ;-)